### PR TITLE
Add a deps target for CI 

### PR DESCRIPTION
**Pull request by Itxaka**
This target will always try to upgrade the packages in the system with
latest versions from our repo

Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

File: make/Makefile.iso [hunk 1]
```diff
@@ -82,7 +82,7 @@ else
 endif
 
 .PHONY: iso
-iso: $(LUET) $(YQ) $(MAKEISO) $(MKSQUASHFS)
+iso: $(LUET) yq $(MAKEISO) $(MKSQUASHFS)
```
Comment (davidcassany) [inline make/Makefile.iso:85]: I guess `$(MAKEISO)` here should be substituted to `makeiso` right?

**Review by davidcassany — APPROVED** [review]
LGTM, just a little leftover I think

File: make/Makefile.iso (hunk 1 above)
Comment (Itxaka) [reply make/Makefile.iso:85]: Good catch!

**Comment by kkaempf** [comment]
Not sure if this is the right approach. The luet installs might also override system binaries (coming e.g. from RPM).
While this PR solves one problem, I fear it might trigger many others.

**Comment by Itxaka** [comment]
> Not sure if this is the right approach. The luet installs might also override system binaries (coming e.g. from RPM).
While this PR solves one problem, I fear it might trigger many others.

I guess this affects building on OBS? How does obs tackles dependency? Not calling luet install, Im sure.

Well, we can do something else, keep the make deps as usual, and have a make deps_ci which does what this patch does. Would that work for you @kkaempf ?

**Comment by kkaempf** [comment]
> I guess this affects building on OBS?
No, not at all.
This affects a local system where binaries are installed from RPM or DEP. Just 'blindly' calling "luet install ..." now likely clobbers these installs.

> 
> Well, we can do something else, keep the make deps as usual, and have a make deps_ci which does what this patch does. 

That should work, thanks.

**Comment by davidcassany** [comment]
> Not sure if this is the right approach. 

On second thought I think this a good point. We should limit this to the toolchain we provide in  `packages/toolchain` so I would exclude `jq` from this PR. 

> The luet installs might also override system binaries (coming e.g. from RPM).

This is a good point I think it should be considered a bug in our toolchain packages, the luet packages should not interfere with whatever the system provides. However this is hard to solve in a sane way, a common solution would be to set luet packages to install under `/usr/local/` buit this collides with our immutable layout approach. So I am not sure there is any other way to make sure we minimize the use of luet packages and ensure those are not provided neither conflict with the system.

**Comment by Itxaka** [comment]
ok, this should work, this only runs on our CI and tries to upgrade all packages in the deps. Should be safe enough to run only on our ci while leaving the make deps target untouched so it can be used locally or on OBS.

**Review by davidcassany — APPROVED** [review]
Looks good

**Pull request merged**